### PR TITLE
Add native configuration repository support library

### PR DESCRIPTION
**Pull request by melix**
This relates to https://github.com/graalvm/native-build-tools/issues/197 by providing a support library which can be used by both the Maven and Gradle plugins.

The library assumes that the repository is available locally on the file system.

The repository can be queried this way:

```
Set<Path> configDirs = repository.findConfigurationDirectoriesFor("my.awesome:library:1.1")
```

or, with more advanced queries:

```
Set<Path> configDirs = repository.findConfigurationDirectoriesFor(spec -> {
    spec.useLatestConfigWhenVersionIsUntested();
    spec.forArtifacts("my.awesome.library:1.1");
    spec.forArtifact(artifact -> {
        artifact.gav("my.other:lib:1.5");
        artifact.forceConfigVersion("5");
    });
});
```

### CODE REVIEW

File: common/native-config/src/main/java/org/graalvm/nativeconfig/internal/DefaultArtifactQuery.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    private String groupId;
+    private String artifactId;
+    private String version;
+
+    private boolean useLatestVersion = false;
+    private String forcedConfig = null;
+
+    @Override
+    public void gav(String gavCoordinates) {
+        String[] gav = gavCoordinates.split(":");
```
Comment (graemerocher) [inline common/native-config/src/main/java/org/graalvm/nativeconfig/internal/DefaultArtifactQuery.java:58]: Do we need to support Maven classifiers?

Comment (melix) [reply common/native-config/src/main/java/org/graalvm/nativeconfig/internal/DefaultArtifactQuery.java:58]: Currently the repository layout doesn't support this. I'd say let's see if a single configuration for all artifacts work (I suspect it won't long term, but we might face other problems first).

**Review by sdeleuze — COMMENTED** [review]
LGTM, should this PR be adapted to take in account the new name `jvm-reachability-metadata` by replacing "Configuration" by "Metadata" and "Native" by "Reachability" for example?

**Review by graemerocher — APPROVED** [review]

**Comment by melix** [comment]
Renames have been done as part of the other PR, to avoid merge conflicts: https://github.com/graalvm/native-build-tools/pull/204/commits/7df6adc80e41609a48db651abe06bde6c0d669b5

**Comment by melix** [comment]
Superceded by #204

**Pull request closed without merging**